### ISSUE TITLE
KeyError "start_time" with interchange.py

### ISSUE THREAD

**Issue by marinerhemant** [Parsl/parsl#3564]
**interchange.py KeyError**
On starting an app from a script, parsl gives an error stating 

```
"interchange.py", line 413, in process_task_outgoing_incoming
'start_time': msg['start_time'],
KeyError: 'start_time'
```

The code then hangs and does nothing.

This is after a fresh install of Anaconda, then installing parsl

Python version: 3.12.4
Parsl version: 2024.07.29

This was executed on a RHEL 8.8.

**Comment by marinerhemant** [comment]
Never mind, fixed it by reverting to parsl==2024.03.18 and deleting all previous python environments.

**Comment by benclifford** [comment]
@marinerhemant this sounds like mixed up versions between your workers (process worker pool) and your submit side Parsl installation.

That start_time field was introduced in PR #3546 which was released in Parsl `2024.07.29` - so if you are using that version on your submit side, but your workers are running an older version (for example, from a different Python environment with a different Parsl version) - then this is what I would expect to happen.

Either of the things you did sounds like they would fix this, because:
* deleting all python environments would stop you accidentally getting the wrong version
* moving back to an earlier version would probably work by chance even if you're getting a different worker version

But you should be able to use the latest Parsl as long as that's what you have installed on the worker side too (there's no protocol compatibility between workers of one Parsl version and submit side of another Parsl version although it often works if the protocol happens to have not changed between the versions)
